### ISSUE TITLE
Better support for custom content types

### ISSUE THREAD

**Issue by artoliukkonen** [kadamwhite/wordpress-rest-api#138]
Currently only way to query custom content types is with .root method.
 
```
/**
 * Generate a query against an arbitrary path on the current endpoint. This is useful for
 * requesting resources at custom WP-API endpoints, such as WooCommerce's `/products`.
 *
 * @method root
 * @param {String} [relativePath] An endpoint-relative path to which to bind the request
 * @param {Boolean} [collection] Whether to return a CollectionRequest or a vanilla WPRequest
 * @return {CollectionRequest|WPRequest} A request object
 */
```

It'd be great to be able to define your own custom content types so one would be able to use those just like .posts and .pages . I have some ideas how to achieve this, will try to do a pull request soon.

**Comment by kadamwhite** [comment]
@artoliukkonen We're discussing options for this in #144 and related issues

**Comment by kadamwhite** [comment]
`.registerRoute` (#176, merged) and Auto-discovery (#74, WIP) are the resolution for this, and those issues supersede this one.
